### ISSUE TITLE
Mixer metric and rule definitions disappeared from functioning deployment

### ISSUE THREAD

**Issue by osowski** [istio/issues#135]
**Is this a BUG or FEATURE REQUEST?:** BUG

**Did you review existing epics or issues to identify if this already being worked on?** (please try to add the correct labels and epics): Yes

**Bug:**
Y/N

**What Version of Istio and Kubernetes are you using, where did you get Istio from, Installation details**
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.3", GitCommit:"f0efb3cb883751c5ffdbe6d515f3cb4fbe7b7acd", GitTreeState:"clean", BuildDate:"2017-11-09T07:27:47Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.2-2+d68ca389c3219c", GitCommit:"d68ca389c3219c63c141a37392f87e579795212b", GitTreeState:"clean", BuildDate:"2017-11-16T08:14:30Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
$ istioctl version
Version: 0.2.12
GitRevision: 998e0e00d375688bcb2af042fc81a60ce5264009
GitBranch: release-0.2
User: releng@0d29a2c0d15f
GolangVersion: go1.8

```

**Is Istio Auth enabled or not ?**
`kubectl apply -f install/kubernetes/istio.yaml`

**What happened**: Following the [Istio Workshop](https://github.com/szihai/istio-workshop), I ran through everything and got to the end successfully.  Istio Dashboard worked, routing rules worked, denial rules worked.  I went to work on other stuff and came back to show the Istio Dashboard again, but all the Mixer-based metrics had disappeared.  I now only had metrics that were defined in the [Workshop Step 8](https://github.com/szihai/istio-workshop/blob/master/exercise-8/README.md).

```
$ kubectl get prometheus,rule -n istio-system
NAME                         AGE
prometheuses/doublehandler   14h

NAME                AGE
rules/doubleprom    14h
rules/newlogstdio   14h
```

I was able to work around this issue by re-running `kubectl apply -f istio.yaml`, which had the following output.  Note that everything besides Telemetry objects seemed to have persisted, while `metric`s and `rule`s were recreated.

```
ravnica:istio-workshop osowski$ kubectl apply -f istio-0.2.12/install/kubernetes/istio.yaml
namespace "istio-system" configured
clusterrole "istio-pilot-istio-system" configured
clusterrole "istio-initializer-istio-system" configured
clusterrole "istio-mixer-istio-system" configured
clusterrole "istio-ca-istio-system" configured
clusterrole "istio-sidecar-istio-system" configured
clusterrolebinding "istio-pilot-admin-role-binding-istio-system" configured
clusterrolebinding "istio-initializer-admin-role-binding-istio-system" configured
clusterrolebinding "istio-ca-role-binding-istio-system" configured
clusterrolebinding "istio-ingress-admin-role-binding-istio-system" configured
clusterrolebinding "istio-egress-admin-role-binding-istio-system" configured
clusterrolebinding "istio-sidecar-role-binding-istio-system" configured
clusterrolebinding "istio-mixer-admin-role-binding-istio-system" configured
configmap "istio-mixer" configured
service "istio-mixer" configured
serviceaccount "istio-mixer-service-account" configured
deployment "istio-mixer" configured
customresourcedefinition "rules.config.istio.io" configured
customresourcedefinition "attributemanifests.config.istio.io" configured
customresourcedefinition "deniers.config.istio.io" configured
customresourcedefinition "listcheckers.config.istio.io" configured
customresourcedefinition "memquotas.config.istio.io" configured
customresourcedefinition "noops.config.istio.io" configured
customresourcedefinition "prometheuses.config.istio.io" configured
customresourcedefinition "stackdrivers.config.istio.io" configured
customresourcedefinition "statsds.config.istio.io" configured
customresourcedefinition "stdios.config.istio.io" configured
customresourcedefinition "svcctrls.config.istio.io" configured
customresourcedefinition "checknothings.config.istio.io" configured
customresourcedefinition "listentries.config.istio.io" configured
customresourcedefinition "logentries.config.istio.io" configured
customresourcedefinition "metrics.config.istio.io" configured
customresourcedefinition "quotas.config.istio.io" configured
customresourcedefinition "reportnothings.config.istio.io" configured
attributemanifest "istioproxy" configured
attributemanifest "kubernetes" configured
stdio "handler" created
logentry "accesslog" created
rule "stdio" created
metric "requestcount" created
metric "requestduration" created
metric "requestsize" created
metric "responsesize" created
metric "tcpbytesent" created
metric "tcpbytereceived" created
prometheus "handler" created
rule "promhttp" created
rule "promtcp" created
configmap "istio" configured
customresourcedefinition "destinationpolicies.config.istio.io" configured
customresourcedefinition "egressrules.config.istio.io" configured
customresourcedefinition "routerules.config.istio.io" configured
service "istio-pilot" configured
serviceaccount "istio-pilot-service-account" configured
deployment "istio-pilot" configured
service "istio-ingress" configured
serviceaccount "istio-ingress-service-account" configured
deployment "istio-ingress" configured
service "istio-egress" configured
serviceaccount "istio-egress-service-account" configured
deployment "istio-egress" configured
serviceaccount "istio-ca-service-account" configured
deployment "istio-ca" configured
```

**What you expected to happen:** I would have expected the Mixer metrics to persist, even after anything happens to Mixer pod itself.

**How to reproduce it:**
Unknown

**Comment by linsun** [comment]
I'm very sure this is exactly what I hit in Oct and discussed on #general and #mixer channels.  The workaround is reinstall mixer which is not pretty.  

@douglas-reid @jmuk thoughts?  Is there anything we should be collecting to help debug this?

**Comment by jmuk** [comment]
Did you happen to use `istioctl delete` to delete some mixer configs? There is a silly bug on `istioctl` which deletes unrelated config entities. If so, that's the case.

We fixed this bug on master. Until the next version is released, please use `kubectl delete` for the time being. Sorry.

**Comment by ldemailly** [comment]
@jmuk thanks ! what's the commit/pr so people can check which build would have the fix (also curious what the bug was)

**Comment by linsun** [comment]
also interested in knowing if this is fixed in 0.3 since i have a 0.3 env.

@osowski see above ^^

**Comment by douglas-reid** [comment]
I believe this is resolved in recent version of Istio. Closing.
